sympy/solvers/ode.py

```diff
@@ -2883,7 +2883,7 @@ def _conditional_term_factoring(expr):
     return expr
 
 
-def constant_renumber(eq, expr):
+def constant_renumber(expr, nonconstants=None, newconstants=None):
```

Comment:
nonconstants -> variables?

Reply:
odesymbols?

Reply:
I prefer a generic name since this function could be used by anyone; it has use beyond ODE.

Reply:
Can you give an example of somewhere else it could be used? That might help with a name. Variables doesn't seem right in this particular context.

Reply:
Also if it has more general use it should probably be moved somewhere else.

Reply:
> Can you give an example of somewhere else it could be used?

If you have `f(x, a, b, c) = a*(b + c*x)` and you want to model that as a function of 'x' with the fewest symbols possible you would want to collapse constants together like `constantsimp` does and the process would have nothing to do with odes. The model could be `c1 + c2*x` in this particular case with `c1 = a*b` and `c2 = a*c`.

Reply:
cf #4435

Reply:
That makes sense but the frontend for that should be something that also calls constantsimp. For now these inputs to the function are sufficient to correctly do its job for dsolve.